geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/z3/XZ3WritableIndex.scala

```diff
@@ -69,7 +69,11 @@ trait XZ3WritableIndex extends AccumuloWritableIndex {
                        (wf: AccumuloFeature): Array[Byte] = {
     val numSplits = splitArray.length
     val split = splitArray(wf.idHash % numSplits)
-    val envelope = wf.feature.getDefaultGeometry.asInstanceOf[Geometry].getEnvelopeInternal
+    val geom = wf.feature.getDefaultGeometry.asInstanceOf[Geometry]
+    if (geom == null) {
+      throw new IllegalArgumentException(s"Null geometry in feature ${wf.feature.getID}")
```

Comment:
Well, third parties can do as they like.  After you rebase, I'd say we can shippit!